### PR TITLE
Fix login API for MongoDB

### LINKED ISSUE #1401: User login API using MongoDB throws TypeError
Environment: Linux - Ubuntu 20.04

Flask-Appbuilder version: 2.3.4

pip freeze output:
apispec==1.3.3
appdirs==1.4.3
astroid==2.4.2
attrs==19.3.0
autopep8==1.5.3
Babel==2.8.0
CacheControl==0.12.6
certifi==2019.11.28
chardet==3.0.4
click==7.1.2
colorama==0.4.3
contextlib2==0.6.0
defusedxml==0.6.0
distlib==0.3.0
distro==1.4.0
dnspython==1.16.0
email-validator==1.1.1
Flask==1.1.2
Flask-AppBuilder==2.3.4
Flask-Babel==1.0.0
Flask-Cors==3.0.8
Flask-JWT-Extended==3.24.1
Flask-Login==0.4.1
flask-mongoengine==0.9.5
Flask-OpenID==1.2.5
Flask-SQLAlchemy==2.4.3
Flask-WTF==0.14.3
html5lib==1.0.1
idna==2.8
ipaddr==2.2.0
isort==4.3.21
itsdangerous==1.1.0
Jinja2==2.11.2
jsonschema==3.2.0
lazy-object-proxy==1.4.3
lockfile==0.12.2
MarkupSafe==1.1.1
marshmallow==2.21.0
marshmallow-enum==1.5.1
marshmallow-sqlalchemy==0.23.1
mccabe==0.6.1
mongoengine==0.20.0
msgpack==0.6.2
numpy==1.18.5
packaging==20.3
pandas==1.0.4
pep517==0.8.2
prison==0.1.3
progress==1.5
pycodestyle==2.6.0
PyJWT==1.7.1
pylint==2.5.3
pymongo==3.10.1
pyparsing==2.4.6
pyrsistent==0.16.0
python-dateutil==2.8.1
python3-openid==3.1.0
pytoml==0.1.21
pytz==2020.1
PyYAML==5.3.1
requests==2.22.0
retrying==1.3.3
six==1.14.0
SQLAlchemy==1.3.17
SQLAlchemy-Utils==0.36.6
toml==0.10.1
urllib3==1.25.8
webencodings==0.5.1
Werkzeug==1.0.1
wrapt==1.12.1
WTForms==2.3.1

Expected results: Access token

2020-06-15 20:22:39,331:ERROR:root:Object of type ObjectId is not JSON serializable
Traceback (most recent call last):
  File "/project/venv/lib/python3.8/site-packages/flask_appbuilder/api/__init__.py", line 83, in wraps
    return f(self, *args, **kwargs)
  File "/project/venv/lib/python3.8/site-packages/flask_appbuilder/security/api.py", line 100, in login
    resp[API_SECURITY_ACCESS_TOKEN_KEY] = create_access_token(
  File "/project/venv/lib/python3.8/site-packages/flask_jwt_extended/utils.py", line 172, in create_access_token
    return jwt_manager._create_access_token(identity, fresh, expires_delta, user_claims,
  File "/project/venv/lib/python3.8/site-packages/flask_jwt_extended/jwt_manager.py", line 511, in _create_access_token
    access_token = encode_access_token(
  File "/project/venv/lib/python3.8/site-packages/flask_jwt_extended/tokens.py", line 76, in encode_access_token
    return _encode_jwt(token_data, expires_delta, secret, algorithm,
  File "/project/venv/lib/python3.8/site-packages/flask_jwt_extended/tokens.py", line 29, in _encode_jwt
    encoded_token = jwt.encode(token_data, secret, algorithm,
  File "/project/venv/lib/python3.8/site-packages/jwt/api_jwt.py", line 58, in encode
    json_payload = json.dumps(
  File "/usr/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/project/venv/lib/python3.8/site-packages/flask_mongoengine/json.py", line 19, in default
    return superclass.default(self, obj)
  File "/project/venv/lib/python3.8/site-packages/flask/json/__init__.py", line 100, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ObjectId is not JSON serializable
2020-06-15 20:22:39,334:INFO:werkzeug:127.0.0.1 - - [15/Jun/2020 20:22:39] "POST /api/v1/security/login HTTP/1.1" 500 -

### PR DESCRIPTION
**Pull request by vinoadth**
Fixes #1401 

Fix login API for MongoDB

### CODE REVIEW

File: flask_appbuilder/security/api.py [hunk 1]
```diff
@@ -95,14 +96,18 @@ def login(self):
         if not user:
             return self.response_401()
 
+        user_id = user.id
+        if isinstance(user_id, ObjectId):
```
Comment (dpgaspar) [inline flask_appbuilder/security/api.py:100]: Can you make this more generic, without the MongoDB dependency at this level

Comment (vinoadth) [reply flask_appbuilder/security/api.py:100]: Okay, let me do it that way...

**Comment by vinoadth** [comment]
please let me know if anything else to be updated

**Comment by yuyou** [comment]
Hi

Has this PR been accepted to the master? It seems not in the Master code where the "user.id" may not be JSON serializable, e.g. in the case of MongoDB.

https://github.com/dpgaspar/Flask-AppBuilder/blob/d04ec3c3399883aa84ed7493c0e2e49160a9f6ec/flask_appbuilder/security/api.py#L111-L117

**Pull request closed without merging**